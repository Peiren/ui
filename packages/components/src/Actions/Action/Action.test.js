import React from 'react';
import { shallow, mount } from 'enzyme';

import Action from './Action.component';

jest.mock('react-dom');
const onClickFn = jest.fn();
const myAction = {
	label: 'Click me',
	icon: 'talend-caret-down',
	onClick: onClickFn,
};

// const myActionWithoutOnClick = {
// 	label: 'Click me',
// 	icon: 'talend-caret-down',
// };


describe('Action', () => {
	it('should render a button', () => {
		// when
		const wrapper = shallow(<Action {...myAction} />);

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should trigger the onclick props when left-click on the button ', () => {
		// given
		const wrapper = shallow(<Action extra="extra" {...myAction} />);
		const buttonWrapper = wrapper.find('Button').at(0);

		// when
		buttonWrapper.simulate('click', { button: 0 });

		// then
		expect(onClickFn).toHaveBeenCalledWith(
			{ button: 0 },
			{action: {extra: 'extra', icon: 'talend-caret-down', label: 'Click me' }, model: undefined});


		// expect(instance.leftClick.mock.calls.length).toBe(1);
		// const args = instance.leftClick.mock.calls[0];
		// expect(args.length).toBe(2);
		// expect(args[0]).toEqual({ button: 0 });
		// expect(args[1].action.extra).toBe('extra');
	});

	// it('should trigger the onclick props when middle-click on the button ', () => {
	// 	// given
	// 	const wrapper = shallow(<Action {...myActionWithoutMockOnClick} />);
	// 	const buttonWrapper = wrapper.find('button').at(0);
	// 	const instance = wrapper.instance();
	//
	// 	// when
	// 	buttonWrapper.simulate('click');
	//
	// 	// then
	// 	expect(instance.onClick).toHaveBeenCalled();
	// 	expect(myAction.onClick.mock.calls.length).toBe(1);
	// 	const args = myAction.onClick.mock.calls[0];
	// 	expect(args.length).toBe(2);
	// 	expect(args[0]).toEqual({ button: 1 });
	// });

	it('should pass all props to the Button', () => {
		// when
		const wrapper = shallow(
			<Action
				className="navbar-btn"
				notExisting
				{...myAction}
			/>);

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should display a Progress indicator if set', () => {
		// when
		const wrapper = shallow(
			<Action
				className="navbar-btn"
				inProgress
				{...myAction}
			/>);

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should display a disabled Icon', () => {
		// when
		const wrapper = shallow(
			<Action
				className="navbar-btn"
				disabled
				{...myAction}
			/>);

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should reverse icon/label', () => {
		// when
		const wrapper = shallow(
			<Action
				iconPosition="right"
				{...myAction}
			/>);

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should apply transformation on icon', () => {
		// when
		const wrapper = shallow(
			<Action
				iconTransform={'rotate-180'}
				{...myAction}
			/>);

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render action with html property name = props.name if set', () => {
		// when
		const wrapper = shallow(
			<Action
				name="custom_name"
				{...myAction}
			/>);

		// then
		expect(wrapper).toMatchSnapshot();
	});
});
