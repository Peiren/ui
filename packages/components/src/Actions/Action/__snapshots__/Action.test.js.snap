// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action should apply transformation on icon 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    role={null}
>
    <Icon
        name="talend-caret-down"
        transform="rotate-180"
    />
    <span>
        Click me
    </span>
</Button>,
  "nodes": Array [
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      role={null}
>
      <Icon
            name="talend-caret-down"
            transform="rotate-180"
      />
      <span>
            Click me
      </span>
</Button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Action
        bsStyle="default"
        disabled={false}
        icon="talend-caret-down"
        iconTransform="rotate-180"
        inProgress={false}
        label="Click me"
        onClick={[Function]}
        tooltipPlacement="top"
/>,
      "_debugID": 13,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "bsStyle": "default",
          "disabled": false,
          "icon": "talend-caret-down",
          "iconTransform": "rotate-180",
          "inProgress": false,
          "label": "Click me",
          "onClick": [Function],
          "tooltipPlacement": "top",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 7,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform="rotate-180"
          />
          <span>
                    Click me
          </span>
</Button>,
        "_debugID": 14,
        "_renderedOutput": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform="rotate-180"
          />
          <span>
                    Click me
          </span>
</Button>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Action
    bsStyle="default"
    disabled={false}
    icon="talend-caret-down"
    iconTransform="rotate-180"
    inProgress={false}
    label="Click me"
    onClick={[Function]}
    tooltipPlacement="top"
/>,
}
`;

exports[`Action should display a Progress indicator if set 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="navbar-btn"
    disabled={true}
    onClick={[Function]}
    onMouseDown={[Function]}
    role={null}
>
    <CircularProgress
        size="small"
    />
    <span>
        Click me
    </span>
</Button>,
  "nodes": Array [
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="navbar-btn"
      disabled={true}
      onClick={[Function]}
      onMouseDown={[Function]}
      role={null}
>
      <CircularProgress
            size="small"
      />
      <span>
            Click me
      </span>
</Button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Action
        bsStyle="default"
        className="navbar-btn"
        disabled={false}
        icon="talend-caret-down"
        inProgress={true}
        label="Click me"
        onClick={[Function]}
        tooltipPlacement="top"
/>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "bsStyle": "default",
          "className": "navbar-btn",
          "disabled": false,
          "icon": "talend-caret-down",
          "inProgress": true,
          "label": "Click me",
          "onClick": [Function],
          "tooltipPlacement": "top",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="navbar-btn"
          disabled={true}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <CircularProgress
                    size="small"
          />
          <span>
                    Click me
          </span>
</Button>,
        "_debugID": 8,
        "_renderedOutput": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="navbar-btn"
          disabled={true}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <CircularProgress
                    size="small"
          />
          <span>
                    Click me
          </span>
</Button>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Action
    bsStyle="default"
    className="navbar-btn"
    disabled={false}
    icon="talend-caret-down"
    inProgress={true}
    label="Click me"
    onClick={[Function]}
    tooltipPlacement="top"
/>,
}
`;

exports[`Action should display a disabled Icon 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="navbar-btn"
    disabled={true}
    onClick={[Function]}
    onMouseDown={[Function]}
    role={null}
>
    <Icon
        name="talend-caret-down"
        transform={undefined}
    />
    <span>
        Click me
    </span>
</Button>,
  "nodes": Array [
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="navbar-btn"
      disabled={true}
      onClick={[Function]}
      onMouseDown={[Function]}
      role={null}
>
      <Icon
            name="talend-caret-down"
            transform={undefined}
      />
      <span>
            Click me
      </span>
</Button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Action
        bsStyle="default"
        className="navbar-btn"
        disabled={true}
        icon="talend-caret-down"
        inProgress={false}
        label="Click me"
        onClick={[Function]}
        tooltipPlacement="top"
/>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "bsStyle": "default",
          "className": "navbar-btn",
          "disabled": true,
          "icon": "talend-caret-down",
          "inProgress": false,
          "label": "Click me",
          "onClick": [Function],
          "tooltipPlacement": "top",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="navbar-btn"
          disabled={true}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
          <span>
                    Click me
          </span>
</Button>,
        "_debugID": 10,
        "_renderedOutput": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="navbar-btn"
          disabled={true}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
          <span>
                    Click me
          </span>
</Button>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Action
    bsStyle="default"
    className="navbar-btn"
    disabled={true}
    icon="talend-caret-down"
    inProgress={false}
    label="Click me"
    onClick={[Function]}
    tooltipPlacement="top"
/>,
}
`;

exports[`Action should pass all props to the Button 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="navbar-btn"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    role={null}
>
    <Icon
        name="talend-caret-down"
        transform={undefined}
    />
    <span>
        Click me
    </span>
</Button>,
  "nodes": Array [
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="navbar-btn"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      role={null}
>
      <Icon
            name="talend-caret-down"
            transform={undefined}
      />
      <span>
            Click me
      </span>
</Button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Action
        bsStyle="default"
        className="navbar-btn"
        disabled={false}
        icon="talend-caret-down"
        inProgress={false}
        label="Click me"
        notExisting={true}
        onClick={[Function]}
        tooltipPlacement="top"
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "bsStyle": "default",
          "className": "navbar-btn",
          "disabled": false,
          "icon": "talend-caret-down",
          "inProgress": false,
          "label": "Click me",
          "notExisting": true,
          "onClick": [Function],
          "tooltipPlacement": "top",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="navbar-btn"
          disabled={false}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
          <span>
                    Click me
          </span>
</Button>,
        "_debugID": 6,
        "_renderedOutput": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="navbar-btn"
          disabled={false}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
          <span>
                    Click me
          </span>
</Button>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Action
    bsStyle="default"
    className="navbar-btn"
    disabled={false}
    icon="talend-caret-down"
    inProgress={false}
    label="Click me"
    notExisting={true}
    onClick={[Function]}
    tooltipPlacement="top"
/>,
}
`;

exports[`Action should render a button 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    role={null}
>
    <Icon
        name="talend-caret-down"
        transform={undefined}
    />
    <span>
        Click me
    </span>
</Button>,
  "nodes": Array [
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      role={null}
>
      <Icon
            name="talend-caret-down"
            transform={undefined}
      />
      <span>
            Click me
      </span>
</Button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Action
        bsStyle="default"
        disabled={false}
        icon="talend-caret-down"
        inProgress={false}
        label="Click me"
        onClick={[Function]}
        tooltipPlacement="top"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "bsStyle": "default",
          "disabled": false,
          "icon": "talend-caret-down",
          "inProgress": false,
          "label": "Click me",
          "onClick": [Function],
          "tooltipPlacement": "top",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
          <span>
                    Click me
          </span>
</Button>,
        "_debugID": 2,
        "_renderedOutput": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
          <span>
                    Click me
          </span>
</Button>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Action
    bsStyle="default"
    disabled={false}
    icon="talend-caret-down"
    inProgress={false}
    label="Click me"
    onClick={[Function]}
    tooltipPlacement="top"
/>,
}
`;

exports[`Action should render action with html property name = props.name if set 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    name="custom_name"
    onClick={[Function]}
    onMouseDown={[Function]}
    role={null}
>
    <Icon
        name="talend-caret-down"
        transform={undefined}
    />
    <span>
        Click me
    </span>
</Button>,
  "nodes": Array [
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      name="custom_name"
      onClick={[Function]}
      onMouseDown={[Function]}
      role={null}
>
      <Icon
            name="talend-caret-down"
            transform={undefined}
      />
      <span>
            Click me
      </span>
</Button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Action
        bsStyle="default"
        disabled={false}
        icon="talend-caret-down"
        inProgress={false}
        label="Click me"
        name="custom_name"
        onClick={[Function]}
        tooltipPlacement="top"
/>,
      "_debugID": 15,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "bsStyle": "default",
          "disabled": false,
          "icon": "talend-caret-down",
          "inProgress": false,
          "label": "Click me",
          "name": "custom_name",
          "onClick": [Function],
          "tooltipPlacement": "top",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 8,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          name="custom_name"
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
          <span>
                    Click me
          </span>
</Button>,
        "_debugID": 16,
        "_renderedOutput": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          name="custom_name"
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
          <span>
                    Click me
          </span>
</Button>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Action
    bsStyle="default"
    disabled={false}
    icon="talend-caret-down"
    inProgress={false}
    label="Click me"
    name="custom_name"
    onClick={[Function]}
    tooltipPlacement="top"
/>,
}
`;

exports[`Action should reverse icon/label 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
    onMouseDown={[Function]}
    role={null}
>
    <span>
        Click me
    </span>
    <Icon
        name="talend-caret-down"
        transform={undefined}
    />
</Button>,
  "nodes": Array [
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      role={null}
>
      <span>
            Click me
      </span>
      <Icon
            name="talend-caret-down"
            transform={undefined}
      />
</Button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Action
        bsStyle="default"
        disabled={false}
        icon="talend-caret-down"
        iconPosition="right"
        inProgress={false}
        label="Click me"
        onClick={[Function]}
        tooltipPlacement="top"
/>,
      "_debugID": 11,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "bsStyle": "default",
          "disabled": false,
          "icon": "talend-caret-down",
          "iconPosition": "right",
          "inProgress": false,
          "label": "Click me",
          "onClick": [Function],
          "tooltipPlacement": "top",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 6,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <span>
                    Click me
          </span>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
</Button>,
        "_debugID": 12,
        "_renderedOutput": <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
          onMouseDown={[Function]}
          role={null}
>
          <span>
                    Click me
          </span>
          <Icon
                    name="talend-caret-down"
                    transform={undefined}
          />
</Button>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Action
    bsStyle="default"
    disabled={false}
    icon="talend-caret-down"
    iconPosition="right"
    inProgress={false}
    label="Click me"
    onClick={[Function]}
    tooltipPlacement="top"
/>,
}
`;
